src/qt/dcclient.cpp

```diff
@@ -126,7 +126,7 @@ wxPaintDCImpl::wxPaintDCImpl( wxDC *owner )
 wxPaintDCImpl::wxPaintDCImpl( wxDC *owner, wxWindow *win )
     : wxWindowDCImpl( owner, win )
 {
-    wxCHECK_RET( m_isWindowPainter,
+    wxCHECK_RET( m_isWindowPainter || win->GetHandle()->inherits("QGLWidget"),
```

Comment:
Using dynamic cast/RTTI like this doesn't look great. I don't know why do we need to allow creating `wxPaintDC` outside of `wxEVT_PAINT` handler for OpenGL windows, but if we really do, I'd add `virtual bool QtCanPaintWithoutActivePainter() const { return false; }` and override it in `wxGLCanvas` to return true.

Reply:
Sorry! I have another idea which makes this change unnecessary. see https://github.com/wxWidgets/wxWidgets/pull/24200#issuecomment-1906787585